client/src/routes/automation/esidom_generator.ts

```diff
@@ -0,0 +1,256 @@
+        json.state = dropdown_sun_sun;
+
+        return JSON.stringify(json);
+    };
+
+    block.binary_condition = (blk: Block) => {
+        const value_service = esidomGenerator.valueToCode(blk, 'Service', PRECEDENCE);
+        const dropdown_state = blk.getFieldValue('state');
+
+        const json: BlocklyJSON = {};
+
+        json.condition = 'state';
+        json.entity_id = value_service;
+        json.state = dropdown_state;
+
+        return JSON.stringify(json);
+    };
+
+    block.color_picker = (blk: Block) => {
+        const color_value = blk.getFieldValue('color');
+
+        console.log(color_value);
+
+        const json: BlocklyJSON = {};
+
+        json.rgb_color = color_value;
+
+        return JSON.stringify(json);
+    };
+
+    block.color_rgb = (blk: Block) => {
+        const number_red = blk.getFieldValue('red');
+        const number_green = blk.getFieldValue('green');
+        const number_blue = blk.getFieldValue('blue');
+        const color_value = `#${number_red.toString(16).padStart(2, '0')}${number_green.toString(16).padStart(2, '0')}${number_blue.toString(16).padStart(2, '0')}`;
+        const json: BlocklyJSON = {};
+
+        json.rgb_color = color_value;
+
+        return JSON.stringify(json);
+    };
+
+    block.time_condition_hour = (blk: Block) => {
+        const number_hour_debut = blk.getFieldValue('Hour_debut');
+        const number_minute_debut = blk.getFieldValue('Minute_debut');
+        const number_second_debut = blk.getFieldValue('Second_debut');
+        const number_hour_end = blk.getFieldValue('Hour_end');
+        const number_minute_end = blk.getFieldValue('Minute_end');
+        const number_second_end = blk.getFieldValue('Second_end');
+
+        const json: BlocklyJSON = {};
+        json.condition = 'time';
+        json.after = `${number_hour_debut}:${number_minute_debut}:${number_second_debut}`;
+        json.before = `${number_hour_end}:${number_minute_end}:${number_second_end}`;
+
+        return JSON.stringify(json);
+    };
+
+    block.time_condition_week = (blk: Block) => {
+        const checkbox_mon = blk.getFieldValue('mon') === 'TRUE';
+        const checkbox_tue = blk.getFieldValue('tue') === 'TRUE';
+        const checkbox_wed = blk.getFieldValue('wed') === 'TRUE';
+        const checkbox_thu = blk.getFieldValue('thu') === 'TRUE';
+        const checkbox_fri = blk.getFieldValue('fri') === 'TRUE';
+        const checkbox_sat = blk.getFieldValue('sat') === 'TRUE';
+        const checkbox_sun = blk.getFieldValue('sun') === 'TRUE';
+        const json: BlocklyJSON = {};
+
+        const weekday = [];
+        if (checkbox_mon) { weekday.push('mon'); }
+        if (checkbox_tue) { weekday.push('tue'); }
+        if (checkbox_wed) { weekday.push('wed'); }
+        if (checkbox_thu) { weekday.push('thu'); }
+        if (checkbox_fri) { weekday.push('fri'); }
+        if (checkbox_sat) { weekday.push('sat'); }
+        if (checkbox_sun) { weekday.push('sun'); }
```

Comment:
This code is duplicated multiple times in this file. Can't we create a function that will perform this operation, and call it instead of repeating the code ? (we could send it the blk object, and it will return the weekday array)
Btw, we could probably do the same when retrieving 'number_hour_debut' & co

Reply:
fixed